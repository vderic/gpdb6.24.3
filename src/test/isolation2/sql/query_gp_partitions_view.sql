-- Test that pg_partitions view does not lock a table.
-- 6X used to acquire lock on partition tables when query
-- gp_partitions view. See `expr_has_vars` in ruleutils.c
-- for more details.
create table pg_partitions_ddl_tab(a int, b int)
distributed by (a) partition by range (b)
( start (1) end (20) every(5::float) );

1: begin;
1: lock pg_partitions_ddl_tab in access exclusive mode;
1: lock pg_partitions_ddl_tab_1_prt_2 in access exclusive mode;


-- The following query should not be blocked by session 1 because no
-- locks should be held by pg_partitions view.
select tablename, partitiontablename, partitionboundary from pg_partitions where tablename like 'pg_partitions_ddl%';

1: end;
1q:
