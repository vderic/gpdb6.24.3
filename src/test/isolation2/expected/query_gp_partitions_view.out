-- Test that pg_partitions view does not lock a table.
-- 6X used to acquire lock on partition tables when query
-- gp_partitions view. See `expr_has_vars` in ruleutils.c
-- for more details.
create table pg_partitions_ddl_tab(a int, b int) distributed by (a) partition by range (b) ( start (1) end (20) every(5::float) );
CREATE

1: begin;
BEGIN
1: lock pg_partitions_ddl_tab in access exclusive mode;
LOCK
1: lock pg_partitions_ddl_tab_1_prt_2 in access exclusive mode;
LOCK


-- The following query should not be blocked by session 1 because no
-- locks should be held by pg_partitions view.
select tablename, partitiontablename, partitionboundary from pg_partitions where tablename like 'pg_partitions_ddl%';
 tablename             | partitiontablename            | partitionboundary                                 
-----------------------+-------------------------------+---------------------------------------------------
 pg_partitions_ddl_tab | pg_partitions_ddl_tab_1_prt_1 | START (1) END (6) EVERY ('5'::double precision)   
 pg_partitions_ddl_tab | pg_partitions_ddl_tab_1_prt_2 | START (6) END (11) EVERY ('5'::double precision)  
 pg_partitions_ddl_tab | pg_partitions_ddl_tab_1_prt_3 | START (11) END (16) EVERY ('5'::double precision) 
 pg_partitions_ddl_tab | pg_partitions_ddl_tab_1_prt_4 | START (16) END (20) EVERY ('5'::double precision) 
(4 rows)

1: end;
END
1q: ... <quitting>
